torch/csrc/jit/graph_executor.cpp

```diff
@@ -477,7 +477,11 @@ void GraphExecutorImplBase::run(Stack& stack) {
   logging::getLogger()->addStatValue(
       logging::runtime_counters::GRAPH_EXECUTOR_INVOCATIONS, 1.0);
 
+#ifdef USE_STATIC_DISPATCH
```

Comment:
Could you please add some comments about why we want to do this?

I know usually we don't want to introduce new source code macros to avoid complicated logic. But for this specific case I'm not entirely sure whether we should reuse USE_STATIC_DISPATCH macro before we measure its impact on perf / compatibility / etc. Note that some production experiment is using full JIT + passes, this change will affect the experiment as well (without gating).

My opinion is to introduce a new source code macro dedicated for this and document its purpose clearly. We can make it clear that it's temporary solution for custom build purpose and will go away eventually after we move to light-interpreter. Then we can set this macro in master CMakeLists.txt when "SELECTED_OP_LIST" option is not empty.

@zdevito - what's your opinion?

Reply:
Took a closer look - seems there already is an exposed c++ API that can toggle the optimization flag:
torch::jit::GraphOptimizerEnabledGuard

I added the guard to the Android wrapper around loading model and calling forward() - it works fine now.
If we don't have plan to remove this API in near future then it seems to be the least intrusive change to fix this problem?
The downside is that for iOS developers they need add the guard manually for now as we don't have a iOS wrapper yet.

Reply:
@ljk53 good finding! I didn't notice there's GraphOptimizerEnabledGuard. My preference is to use that in Android wrapper. This PR can be canceled then.

Reply:
With this guard, it skips most passes but still runs some "required passes", above which the comment says:
```
    // Phase 1. Specialize to input definedness (this is very important for gradient graphs), and run required passes to bring the graph to an executable form.
```
I'm not sure whether the "required passes" can change operators. On the other hand they also seem to be very critical to not skip.

Reply:
In runRequiredPasses(g), there's CanonicalizeOps(g) that might change ops?

Reply:
I took a brief look at jit/passes/canonicalize_ops.cpp, seems it can insert new node of "aten::add" / "aten::sub" / "aten::mul" / "aten::div" if such operator exists in the graph and its input tensor dimension is 0 - in which case it will replace tensor input with constant. Another change is for "aten::chunk" it will do some "setType" calls on its output for certain cases.
From surface it seems not adding brand new operators. And after reading the logic I actually feel it might affect the correctness of the call if we skip the required passes...

Reply:
Looks like CanonicalizeOps() only inserts prim::constant/chunk. I assume it's safe?

Reply:
In the 2nd case it might insert "prim::ConstantChunk" op but we include all prim ops for now.

Reply:
As long as all prim ops are included I think we are safe. Which part do you think if we don't do required passes it would affect the correctness?

Reply:
> As long as all prim ops are included I think we are safe. Which part do you think if we don't do required passes it would affect the correctness?

E.g.: replacing zero-dim tensor with constant - I'm not entirely sure whether it does so just for following optimization passes or to make it well-formed (as zero dim tensor might break some downstream invariant in operator library)?